docs/utilities/idempotency.md

```diff
@@ -0,0 +1,351 @@
+---
+title: Idempotency
+description: Utility
+---
+
+This utility provides a simple solution to convert your Lambda functions into idempotent operations which are safe to
+retry.
+
+## Terminology
+
+The property of idempotency means that an operation does not cause additional side effects if it is called more than
+once with the same input parameters. Idempotent operations will return the same result when they are called multiple
+times with the same parameters. This makes idempotent operations safe to retry.
+
+
+## Key features
+
+* Prevent Lambda handler code executing more than once on the same event payload during a time window
+* Ensure Lambda handler returns the same result when called with the same payload
+* Select a subset of the event as the idempotency key using JMESpath expressions
+* Set a time window in which records with the same payload should be considered duplicates
+
+## Getting started
+
+### Required resources
+
+Before getting started, you need to create a DynamoDB table to store state used by the idempotency utility. Your lambda
```

Comment:
Should we put this as the general case? I can quickly see an ask for ElastiCache in the future for example, so wouldn't it be better to put it under a DynamoDB section?

Reply:
You're right, but I'll save a bigger restructuring of the docs until we have more support for other persistence layers.